Packs/AnythingLLM/Integrations/AnythingLLM/AnythingLLM.py

```diff
@@ -343,45 +423,51 @@ def _list(self, items: str):
 
         return response
 
-
     def _embedding(self, workspace: str, folder: str, document: str, action: str):
         try:
+            name = ""
             name = document_name(folder, document, self.document_list())
 
             try:
                 ws = self.workspace_get(workspace)
             except Exception:
-                raise Exception(f"workspace [{workspace}] not found")
+                raise Exception(f"workspace not found")
 
             if action == "adds":
-                if embedding_exists(name, ws):
-                    raise Exception(f"[{document}] already embedded")
+                if embedding_exists(ws, document):
+                    raise Exception(f"already embedded")
             elif action == "deletes":
-                if not embedding_exists(name, ws):
-                    raise Exception(f"[{document}] not embedded")
+                if not embedding_exists(ws, document):
+                    raise Exception(f"not embedded")
+            else:
+                raise Exception(f"action [{action}] not 'adds' or 'deletes' ")
 
             data = {
                 action: [f"{folder}/{name}"]
             }
             slug = workspace_slug(workspace, self.workspace_list())
             response = self._http_request(
-                method = "POST",
-                url_suffix = f"/v1/workspace/{slug}/update-embeddings",
-                json_data = data
+                method="POST",
+                url_suffix=f"/v1/workspace/{slug}/update-embeddings",
+                json_data=data
             )
         except Exception as e:
-            msg = f"AnythingLLM: _embedding: exception [{action}] a document embedding - {str(e)}"
+            msg = f"AnythingLLM: _embedding: exception [{action}] a document embedding [{document}] in [{workspace}] - {str(e)}"
             demisto.debug(msg)
             raise Exception(msg)
 
         return response
 
+
 ''' HELPER FUNCTIONS '''
 
-def embedding_exists(docname: str, ws: dict) -> bool:
-    for doc in ws['workspace']['documents']:
-        if doc['filename'] == docname:
-            return True
+
+def embedding_exists(ws: dict, document: str) -> bool:
+    if "documents" in ws['workspace'][0]:
+        for doc in ws['workspace'][0]['documents']:
+            metadata = json.loads(doc['metadata'])
+            if metadata['title'] == document:
+                return True
```

Comment:
What if there is no workspace in the dictionary? This call would lead to a key error

Reply:
This is an internal function - workspace (**ws**) is already validated before it is invoked